components/eam/src/chemistry/mozart/mo_drydep.F90

```diff
@@ -2523,702 +975,682 @@ subroutine drydep_xactive( ncdate, sfc_temp, pressure_sfc,  &
+       thg(icol) = get_potential_temperature(sfc_temp(icol), pressure_sfc(icol), spec_hum(icol))
+
        !-------------------------------------------------------------------------------------
        ! height of 1st level
        !-------------------------------------------------------------------------------------
-       z(i) = - r/grav * air_temp(i) * (1._r8 + .61_r8*spec_hum(i)) * log(p/pg)
+       zl(icol) = - r/grav * air_temp(icol) * (1._r8 + .61_r8*spec_hum(icol)) * log(pressure_10m(icol)/pressure_sfc(icol))
        !-------------------------------------------------------------------------------------
        ! wind speed
        !-------------------------------------------------------------------------------------
-       va(i) = max( .01_r8,wind_speed(i) )
+       va(icol) = max( .01_r8,wind_speed(icol) )
+
        !-------------------------------------------------------------------------------------
        ! Richardson number
        !-------------------------------------------------------------------------------------
-       ribn(i) = z(i) * grav * (tha(i) - thg(i))/thg(i) / (va(i)*va(i))
-       ribn(i) = min( ribn(i),ric )
-       unstable(i) = ribn(i) < 0._r8
+       ribn(icol) = zl(icol) * grav * (tha(icol) - thg(icol))/thg(icol) / (va(icol)*va(icol))
+       ribn(icol) = min( ribn(icol),ric )
+
+       unstable(icol) = ribn(icol) < 0._r8
        !-------------------------------------------------------------------------------------
-       ! saturation vapor pressure (Pascals)
-       ! saturation mixing ratio
        ! saturation specific humidity
        !-------------------------------------------------------------------------------------
-       es    = 611._r8*exp( 5414.77_r8*(sfc_temp(i) - tmelt)/(tmelt*sfc_temp(i)) )
-       ws    = .622_r8*es/(pg - es)
-       qs(i) = ws/(1._r8 + ws)
-       has_dew(i) = .false.
-       if( qs(i) <= spec_hum(i) ) then
-          has_dew(i) = .true.
-       end if
-       if( sfc_temp(i) < tmelt ) then
-          has_dew(i) = .false.
-       end if
+       qs(icol) = get_saturation_specific_humidity(sfc_temp(icol), pressure_sfc(icol))
+     
+       has_dew(icol) = .false.
+       if( qs(icol) <= spec_hum(icol) ) then
+          has_dew(icol) = .true.
+       endif
+       if( sfc_temp(icol) < tmelt ) then
+          has_dew(icol) = .false.
+       endif
        !-------------------------------------------------------------------------------------
        ! constant in determining rs
        !-------------------------------------------------------------------------------------
-       tc(i) = sfc_temp(i) - tmelt
-       if( sfc_temp(i) > tmelt .and. sfc_temp(i) < 313.15_r8 ) then
-          crs(i) = (1._r8 + (200._r8/(solar_flux(i) + .1_r8))**2) * (400._r8/(tc(i)*(40._r8 - tc(i))))
+       tc(icol) = sfc_temp(icol) - tmelt
+       if( sfc_temp(icol) > tmelt .and. sfc_temp(icol) < 313.15_r8 ) then
```

Comment:
I'm not sure exactly, but 313.15 K = 40.0 C assuming tmelt = 273.15.   If sfc_temp exceeds this, then 40._r8 - tc in the next expression becomes negative (and if sfc_temp == 313.15_r8, then the next expression divides by zero).   Ideally it seems there should be some constant such as 'tcmax = 40._r8' that is used in the next expression, along with the test condition 'if( tc(icol) > 0._r8 and. tc(icol) < tcmax ) then' -- but I'm not sure if that would break BFB.

Reply:
I think what @brigaudet said makes sense. It is the upper bound of temperature applicable for the below calculation. I just made it a parameter.